src/zenml/integrations/mlflow/experiment_trackers/mlflow_experiment_tracker.py

```diff
@@ -195,6 +196,18 @@ def prepare_step_run(self, info: "StepRunInfo") -> None:
             experiment_name=experiment_name, run_name=info.run_name
         )
 
+        # Validate that the run exists before attempting to resume it
+        if run_id:
+            try:
+                mlflow.get_run(run_id)
+            except MlflowException as e:
+                # Run doesn't exist on the MLflow server, create a new one
+                logger.warning(
+                    f"Run with id {run_id} not found in MLflow tracking server. "
+                    f"Creating a new run instead. Error: {e}"
+                )
+                run_id = None
+
```

Comment:
I don't expect this MlflowException to ever be raised given that the run is fetched from MLflow itself above with the get_run_id call. If the run does indeed not exist, then the get_run_id implementation is broken.